packages/design-system/src/components/analytics/getAnalyticsContentFromRefs.ts

```diff
@@ -1,7 +1,34 @@
 import { RefObject } from 'react';
 
+/**
+ * Gets text content from an element and even follows `slot` elements across Shadow DOM
+ * barriers to get the full text that the user sees.
+ */
+function getTextContent(nodes: Node[]): string {
+  return nodes
+    .map((node) => {
+      if (node.nodeType === Node.TEXT_NODE) {
+        return node.textContent;
```

Comment:
I had another quick question—could `node.textContent` ever be `undefined` or is it safe to assume it will always return a string in this context?

Reply:
It should be safe to assume it's always at least an empty string for the `TEXT_NODE` type. I'm pretty sure it's part of the spec, and TypeScript thinks it always has a value.